java/arcs/core/storage/DirectStore.kt

```diff
@@ -263,6 +264,19 @@ class DirectStore<Data : CrdtData, Op : CrdtOperation, T> /* internal */ constru
                 throw e
             }
         }
+
+        /**
+         * The [modelChange] is not empty against the first received model from the
```

Comment:
for the first model, [modelChange] is not empty, but [otherChange] should be empty. so noDriverSideChanges=true and updateStateAndAct(true, version, true) should not do anything. Let me dig in on otherChange a bit more.

Reply:
iirc both modelChange and otherChange are not empty.
It could be my fault on writing that comment.

Reply:
yep confirmed the `modelChange` == `otherChange` in the first sync.
I changes the [modelChange] to [otherChange] in the comment, my fault!

Reply:
[otherChange] should be empty. If it is not (and your test demonstrate that) it means there is a bug in CrdtEntity. So I would prefer to fix the bug in crdtentity rather than introducing this workaround. Let me find and fix the bug, and we can verify against this new test.

Reply:
see https://github.com/PolymerLabs/arcs/pull/5886. I think that's a better fix since it address the underlying problem in crdtentity. I copied the test from this pr to show that it fixes the redundant database write.

Reply:
Looks good! Take your solution!